settings_vial/__init__.py

```diff
@@ -0,0 +1,7 @@
+import logging
+
+from .settings_vial import Settings
+
+__all__ = [Settings]
+
+logging.getLogger(__name__).addHandler(logging.NullHandler())
```

Comment:
I don't think this does what you think it is doing.  It turns off default logging to stdout/stderr (https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library).  Also b/c you don't use the logger, but the warning module, the relevant logger would be 'py.warnings' (https://docs.python.org/3.7/library/warnings.html, and https://docs.python.org/3.7/library/logging.html#logging.captureWarnings)

Reply:
Actually, since I can't get captureWarnings to work it is clear I still don't understand something here.

Reply:
You're absolutely right, I chose my words poorly. What I meant was that the library should not handle log events, the user should decide how those are handled. By adding the `NullHandler` as you mentioned, it then does nothing about those events.
By using `__name__` we ensure no naming collisions with the loggers.

From this link: https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library I think the most relevant part is the *Note* box.

An example usage so that you can see the `logging.captureWarnings(True)` working:

```python
import logging
from settings_vial import Settings

logging.captureWarnings(True)
logging.basicConfig(filename='app.log', filemode='w', format='%(name)s - %(levelname)s - %(message)s')

settings = Settings('PREFIX_', 'OVERRIDE_', 'not_callable')
settings.BLA
```
You can read more about the different attributes in the basicConfig format here: https://docs.python.org/3/library/logging.html#logrecord-attributes

This would write the following to `app.log`:
```
py.warnings - WARNING - .../settings-vial/settings_vial/settings_vial.py:67: UnsupportedSetType: Override callable does not return a set or a list
  warnings.warn("Override callable does not return a set or a list", UnsupportedSetType)
```

Reply:
Problem with my testing of `warning` module was that I hadn't read the docs well enough.  Repeat warnings are ignored (https://docs.python.org/3.7/library/warnings.html#the-warnings-filter).